urlb879d9e865_ogotalski_elevator_tgz-pJ8-ElevatorTaskJ8/src/by/epam/lab/ElevatorController.java

```diff
@@ -0,0 +1,219 @@
+	private static final String STARTING_TRANSPORTATION = "STARTING_TRANSPORTATION";
+	private static final String NEW_NEXT_FLOOR = "NEW NEXT_FLOOR ";
+
+	private enum Direction {
+		UP(1), DOWN(-1);
+		private final int intDirection;
+
+		private Direction(int i) {
+			intDirection = i;
+		}
+	}
+
+	private int sleepTime;
+	private final Building building;
+	private final Elevator elevator;
+	private Direction direction;
+	private Floor nextFloor;
+	private int outOnNextFloor;
+	private boolean working;
+	private int notReadyPassengers = 0;
+	private int inOnCurrFloor;
+	
+
+	public synchronized  void ready() {
+		notReadyPassengers--;
+	}
+
+	public void setNotReadyPassengers(int i) {
+		notReadyPassengers = i;
+	}
+	
+	public ElevatorController(Building building, Elevator elevator) {
+		super();
+		this.building = building;
+		this.elevator = elevator;
+		this.nextFloor = elevator.getCurrentFloor();
+		this.outOnNextFloor = 0;
+		this.working = false;
+		this.direction = Direction.UP;
+	}
+
+	public boolean isWorking() {
+		return working;
+	}
+
+	public Building getBuilding() {
+		return building;
+	}
+
+	public Elevator getElevator() {
+		return elevator;
+	}
+
+	public Floor getFloor() {
+		return elevator.getCurrentFloor();
+	}
+
+	public void changeNextFloor() {
+		Floor passengerFloor;
+		nextFloor = null;
+		outOnNextFloor = 0;
+		for (Passenger passenger : elevator.getElevatorContainer()) {
+			passengerFloor = passenger.getDestFloor();
+			if (nextFloor == null)
+				nextFloor = passengerFloor;
+			int i = nextFloor.compareTo(passengerFloor);
+			if (i == 0) {
+				outOnNextFloor++;
+			} else if ((direction.intDirection) * i > 0) {
+				nextFloor = passengerFloor;
+				outOnNextFloor = 1;
+			}
+		}
+		LOG.trace(NEW_NEXT_FLOOR + nextFloor);
+	}
+
+	public boolean addPassenger(Passenger passenger)
+			throws InterruptedException {
+		Floor passengerFloor = passenger.getCurrentFloor();
+
+		synchronized (this) {
+
+			try {
+				if (!elevator.getCurrentFloor().equals(
+						passenger.getCurrentFloor()))
+					return false;
+				inOnCurrFloor--;
+				if (elevator.getCurrentFloor().compareTo(
+						passenger.getDestFloor())
+						* direction.intDirection > 0)
+					return false;
+				if (!elevator.addPassenger(passenger))
+					return false;
+
+				passengerFloor.removeDispatchPassenger(passenger);
+				Thread.sleep(sleepTime);
+
+			} finally {
+				this.notify(); // maybe better notifyAll?
+			}
+		}
+		return true;
+	}
+
+	public boolean removePassenger(Passenger passenger)
+			throws InterruptedException {
+		Floor passengerFloor = passenger.getDestFloor();
+		synchronized (this) {
+			try {
+				if (!elevator.removePassenger(passenger))
+					return false;
+
+				outOnNextFloor--;
+				passengerFloor.addArrivalPassenger(passenger);
+				Thread.sleep(sleepTime);
+			} finally {
+				this.notify();
+			}
+		}
+		return true;
+	}
+
+	public void doJob(int sleepTime) throws InterruptedException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 45 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.